packages/account/src/BiconomySmartAccountV2.ts

```diff
@@ -165,6 +169,32 @@ export class BiconomySmartAccountV2 extends BaseSmartAccount {
    * calculate the account address even before it is deployed
    */
   async getCounterFactualAddress(params?: CounterFactualAddressParam): Promise<string> {
+    const validationModule = params?.validationModule ?? this.defaultValidationModule;
+    const index = params?.index ?? this.index;
+
+    // Note: Review
+    // the fact that below flow (AddressResolver) is Always called can be avoided by passing a flag in smart account config
+    // (if it's intended to detect V1 upgraded accounts)
```

Comment:
need to add this flag

Reply:
added

Reply:
check this commit for this and also above comment.
https://github.com/bcnmy/biconomy-client-sdk/commit/aace8b867167f0bbbd852dab1715aebf3db0864c